transitions/index.js

```diff
@@ -73,24 +103,34 @@ const processChange = (change, callback) => {
  * Load transitions using `require` based on what is in AVAILABLE_TRANSITIONS
  * constant and what is enabled in the `transitions` property in the settings
  * data.  Log warnings on failure.
+ *
+ * Tests can diabled autoEnableSystemTransitions for more control.
  */
-const loadTransitions = () => {
+const loadTransitions = (autoEnableSystemTransitions = true) => {
+
   const self = module.exports;
-  const configuredTransitions = config.get('transitions') || [];
-  _.each(configuredTransitions, (conf, key) => {
-    if (!conf || conf.disable) {
-      return logger.warn(`transition ${key} is disabled`);
-    }
-    if (AVAILABLE_TRANSITIONS.indexOf(key) === -1) {
-      return logger.warn(`transition ${key} not available.`);
+  const transitionsConfig = config.get('transitions') || [];
+
+  // Load all system or configured transitions
+  AVAILABLE_TRANSITIONS.forEach(transition => {
+    const conf = transitionsConfig[transition];
+
+    const system =
+      autoEnableSystemTransitions && SYSTEM_TRANSITIONS.includes(transition);
+
+    const disabled = conf && conf.disable;
+
+    if ((system && disabled) || (!system && !conf || disabled)) {
+      return logger.warn(`transition ${transition} is disabled`);
```

Comment:
Man… this `if` statement honestly took me an hour. No joke. I mean it worked in 5 minutes, but I kept on redoing it to try to make it clearer, because evidently it's a really challenging thing to make readable.

But yes: it is intentional that you can disable system transitions. There is documentation against the `SYSTEM_TRANSITONS` const to that effect.

At least, that was the intention when I first wrote it. Standing back from it now I'm not sure if that's still a relevant thing to do, see my comment on the ticket. I'm going to leave this for now, we can always change it later.